integration-tests/http/__tests__/cart/store/cart.spec.ts

```
@@ -2815,6 +2815,25 @@ medusaIntegrationTestRunner({
           )
         })
 
+        it("should throw an error when adding a promotion that does not exist", async () => {
+          const invalidPromoCode = "SOME_INVALID_PROMO_CODE"
+
+          try {
+            await api.post(
+              `/store/carts/${cart.id}/promotions`,
+              { promo_codes: [invalidPromoCode] },
+              storeHeaders
+            )
+            fail()
+          } catch (e) {
+            expect(e.response.status).toEqual(400)
+            expect(e.response.data.type).toEqual("invalid_data")
+            expect(e.response.data.message).toEqual(
+              `The promotion code ${invalidPromoCode} is invalid`
+            )
+          }
+        })
+
```

suggestion: you can grab the error and check it outside the try catch.

`const { response } = await api.post(...).catch((e) => e)`

generally speaking, ideally we avoid using `try/catch` on tests to avoid hiding errors. And we don't have to safeguard it with `fail`

---

That would explain why I didn't find fail anywhere in the repo haha